generator/src/lib.rs

```diff
@@ -1986,28 +1988,30 @@ fn derive_setters(
                 assert!(field.null_terminate);
                 assert_eq!(field.size, None);
                 return Some(quote! {
-                    #[inline]
                     #deprecated
+                    #[inline]
+                    #[must_use]
                     pub fn #param_ident_short(mut self, #param_ident_short: &'a core::ffi::CStr) -> Self {
                         self.#param_ident = #param_ident_short.as_ptr();
                         self
                     }
-                    #[inline]
                     #deprecated
+                    #[inline]
+                    #[must_use]
                     pub unsafe fn #param_ident_as_c_str(&self) -> &core::ffi::CStr {
                         core::ffi::CStr::from_ptr(self.#param_ident)
                     }
                 });
             } else if is_static_array(field) {
                 assert_eq!(field.size, None);
                 return Some(quote! {
-                    #[inline]
                     #deprecated
+                    #[inline]
                     pub fn #param_ident_short(mut self, #param_ident_short: &core::ffi::CStr) -> core::result::Result<Self, CStrTooLargeForStaticArray> {
```

Comment:
Shouldn't this have it too?

Reply:
Not sure if we can annotate `Self` like that so that `.some_string(CStr...)?;` still counts as dropping `Self` by accident...

But then the same applies to a lacking "`must_use` pattern" on the previously discussed `(Vec<vk::Pipeline>, vk::?Result)` returns...

Reply:
I have contemplated whether such an annotation on the `struct` as a whole might appear too heavy. The standard library doesn't mark many types as such either (But in this case the only usefulness is the returned value, so it makes sense to annotate it as such).

Reply:
It seems like it'll work well for us, and it's low risk.